tests/app/questionnaire/test_router.py

```diff
@@ -395,3 +395,150 @@ def test_get_last_complete_location_in_section(self):
+
+        routing_path = [
+            Location(
+                section_id='who-lives-here-section',
+                block_id='primary-person-list-collector',
+            ),
+            Location(
+                section_id='who-lives-here-section',
+                block_id='anyone-else-list-collector',
+            ),
+            Location(
+                section_id='who-lives-here-section',
+                block_id='who-lives-here-section-summary',
+            ),
+            Location(
+                section_id='who-lives-here-section',
+                block_id='relationships-interstitial',
+            ),
+            Location(section_id='who-lives-here-section', block_id='relationships'),
+        ]
+
+        location_when_section_complete = router.get_location_for_hub_when_section_complete(
+            routing_path=routing_path
+        )
+
+        self.assertEqual(
+            location_when_section_complete,
+            Location(
+                section_id='who-lives-here-section',
+                block_id='who-lives-here-section-summary',
+            ),
+        )
+
+    def test_get_location_for_hub_when_section_complete_section_summary_last_block(
+        self
+    ):
+        schema = load_schema_from_name('test_hub_section_summary')
+        router = Router(schema)
+
+        routing_path = [
+            Location(section_id='accommodation-section', block_id='accommodation-type'),
+            Location(section_id='accommodation-section', block_id='type-of-house'),
+            Location(
+                section_id='accommodation-section',
+                block_id='accommodation-section-summary',
+            ),
+        ]
+        location_when_section_complete = router.get_location_for_hub_when_section_complete(
+            routing_path=routing_path
+        )
+        self.assertEqual(
+            location_when_section_complete,
+            Location(
+                section_id='accommodation-section',
+                block_id='accommodation-section-summary',
+            ),
+        )
+
+    def test_get_location_for_hub_when_section_complete_no_section_summary(self):
+        schema = load_schema_from_name('test_hub_section_summary')
+        router = Router(schema)
+
+        routing_path = [
+            Location(
+                section_id='vehicles-section',
+                block_id='number-of-vehicles-introduction',
+            ),
+            Location(section_id='vehicles-section', block_id='type-of-house'),
+        ]
+
+        location_when_section_complete = router.get_location_for_hub_when_section_complete(
+            routing_path=routing_path
+        )
+
+        self.assertEqual(
+            location_when_section_complete,
+            Location(
+                section_id='vehicles-section',
+                block_id='number-of-vehicles-introduction',
+            ),
+        )
+
+    def test_get_next_location_url_for_hub_when_section_complete(self):
```

Comment:
I think the method name would be clearer as `test_get_next_location_url_for_mid_section_summary_when_section_complete`. I think it's also worth adding a test for when the section isn't complete.